react-ui/src/components/Login/SignIn.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+import Link from '@mui/material/Link';
+import Box from '@mui/material/Box';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { authenticationService } from '../../services/authentication.service';
+import logo from '../../egeria-logo.png';
+
+
+interface Props {
+}
+
+
+interface State {
+
+    isLoading: Boolean,
+    errors: Map<string, boolean>,
+    feedbackMessage: String,
+}
+
+/**
+ *
+ * React component used for displaying login form.
+ *
+ * @since      0.1.0
+ * @access     public
+ *
+ */
+
+const theme = createTheme();
+
+class SignIn extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            isLoading: false,
+            errors: new Map<string, boolean>(),
+            feedbackMessage: 'Sign in',
+        };
+    }
+
+    /**
+     * handles the submit
+     * @param event of the react form
+     */
+    handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const data = new FormData(event.currentTarget);
+        let { errors, feedbackMessage } = this.state;
+        let username = String(data.get('username'));
+        let password = String(data.get('password'));
+
+        errors.set('username',username ==='');
+        errors.set('password', password ==='');
+
+        // check if the form has validation errors
+        let hasErrors = Array.from(errors.values()).find((item) => item);
+
+        if( ! hasErrors ){
+            authenticationService.login( username, password).then((response) => {
+                if (!response.ok) {
+                    switch (response.status){
+                        case 401 : feedbackMessage = "Wrong credentials!"; break;
+                        case 403 : feedbackMessage = "You are not authorized to access this application."; break;
+                        default : feedbackMessage = "Ops! Cannot authenticate right now."; break;
+                    }
+                    this.setState({feedbackMessage});
```

Comment:
this can be moved in the same setState with the one on line 78, you already set the value for the feedbackMessage, the `this.setState` should always be triggered at the end of all logic because it is an async and it triggers a new react component cycle